src/invidious/yt_backend/youtube_api.cr

```diff
@@ -616,12 +616,19 @@ module YoutubeAPI
 
     headers = HTTP::Headers{
       "Content-Type"              => "application/json; charset=UTF-8",
-      "Accept-Encoding"           => "gzip, deflate",
       "x-goog-api-format-version" => "2",
       "x-youtube-client-name"     => client_config.name_proto,
       "x-youtube-client-version"  => client_config.version,
     }
 
+    if CONFIG.invidious_companion && endpoint == "/youtubei/v1/player"
+      headers["Authorization"] = "Bearer " + CONFIG.hmac_key
```

Comment:
ok what do you think of a new parameter key for validating the communication with external programs? could be used not only for invidious companion but any other program that could be integrated with invidious.

like `SECURE_KEY`? I'm not sure what name to come up with that doesn't involve confusion in having two names with `HMAC` in it. I'm thinking about the average user trying to diagnose an issue.

Reply:
Validating the communication is a whole other animal than a simple `bearer` authentication. That's basically what HTTPS/certificates are for!

If your goal is to prevent abuse of the companion if some of its endpoints are not properly protected, a simple API key is enough for that. The simplest option would be to pass each key as part of the companion config, as I explained above.

Reply:
ok then I call it `INVIDIOUS_COMPANION_KEY`. Simple as that. I don't want to make it too complicated.

Letting invidious companion generating its own api key then copying to invidious config is too much. Especially if you have multiple companions, it becomes a nightmare to handle.